### PR TITLE
ADCM-6156: Implement missing methods for BundleNode

### PR DESCRIPTION
**Pull request by Starovoitov**

### CODE REVIEW

File: adcm_aio_client/core/objects/cm.py [hunk 1]
```diff
@@ -81,6 +86,32 @@ def get_own_path(self: Self) -> Endpoint:
...
     class_type = Bundle
 
+    async def _download_external_bundle(self: Self, url: str) -> bytes:
+        try:
+            urlopen(url)  # noqa S310
+            async with httpx.AsyncClient() as client:
+                response = await client.get(url)
+                return response.content
+        except ValueError as err:
+            raise OperationError from err
```
Comment (Sealwing) [inline adcm_aio_client/core/objects/cm.py:100]: cryptic

please explain

File: adcm_aio_client/core/objects/cm.py [hunk 2]
```diff
@@ -81,6 +86,32 @@ def get_own_path(self: Self) -> Endpoint:
...
+            file_content = await self._download_external_bundle(bundle_loc)
+            files = {"file": file_content}
+        else:
+            files = {"file": Path(bundle_loc).read_bytes()}
+
+        response = await self._requester.post("bundles", data=files)
+
+        if accept_license:
+            (await self.get()).license.accept()
+
```
Comment (Sealwing) [inline adcm_aio_client/core/objects/cm.py:113]: 1. No need in defaulting to `None` for accept license, just `accept_license: bool = False`
2. Requirements defines signature for source as `source: UrlPath | Path`, where `Path` is `pathlib.Path` object (so you can check intentions by checking `isinstance(source, Path)`) and `UrlPath` is either string or our custom string wrapper (probably our custom type is better for now, so you can check it's type too). See that argument is named differently in requirements

File: adcm_aio_client/core/objects/cm.py [hunk 3]
```diff
@@ -81,6 +86,32 @@ def get_own_path(self: Self) -> Endpoint:
...
+        if not Path(bundle_loc).exists():
+            file_content = await self._download_external_bundle(bundle_loc)
+            files = {"file": file_content}
+        else:
+            files = {"file": Path(bundle_loc).read_bytes()}
+
+        response = await self._requester.post("bundles", data=files)
+
+        if accept_license:
+            (await self.get()).license.accept()
```
Comment (Sealwing) [inline adcm_aio_client/core/objects/cm.py:112]: build bundle before that, then accept license

you can already have bundle object at this point

File: adcm_aio_client/core/objects/cm.py [hunk 4]
```diff
@@ -81,6 +86,32 @@ def get_own_path(self: Self) -> Endpoint:
...
+
+        return Bundle(requester=self._requester, data=response.as_dict())
+
+    async def delete(self: Self) -> None:
+        await self._requester.delete(*self.get_own_path())
```
Comment (Sealwing) [inline adcm_aio_client/core/objects/cm.py:117]: using `Deletable` mixin is preferred

**Review by Sealwing — CHANGES_REQUESTED** [review]

File: adcm_aio_client/core/objects/cm.py (hunk 1 above)
Comment (Starovoitov) [reply adcm_aio_client/core/objects/cm.py]: urlopen was meant to checj whether the string is real url

File: adcm_aio_client/core/objects/cm.py (hunk 2 above)
Comment (Starovoitov) [reply adcm_aio_client/core/objects/cm.py]: changed

File: adcm_aio_client/core/objects/cm.py (hunk 3 above)
Comment (Starovoitov) [reply adcm_aio_client/core/objects/cm.py]: changed

File: adcm_aio_client/core/objects/cm.py (hunk 4 above)
Comment (Starovoitov) [reply adcm_aio_client/core/objects/cm.py]: changed

File: adcm_aio_client/core/objects/cm.py [hunk 5]
```diff
@@ -85,6 +89,30 @@ def _main_prototype_id(self: Self) -> int:
...
+                return response.content
+        except ValueError as err:
+            raise OperationError from err
+
+    async def create(self: Self, source: Path | UrlPath, accept_license: bool = False) -> Bundle:  # noqa: FBT001, FBT002
+        if isinstance(source, UrlPath):
+            file_content = await self._download_external_bundle(source)
+            files = {"file": file_content}
+        else:
+            files = {"file": Path(source).read_bytes()}
```
Comment (Sealwing) [inline adcm_aio_client/core/objects/cm.py:105]: it's already `Path` here

File: adcm_aio_client/core/objects/cm.py [hunk 6]
```diff
@@ -85,6 +89,30 @@ def _main_prototype_id(self: Self) -> int:
...
+
+        response = await self._requester.post("bundles", data=files)
+
+        bundle = Bundle(requester=self._requester, data=response.as_dict())
+
+        if accept_license:
+            bundle.license.accept()
```
Comment (Sealwing) [inline adcm_aio_client/core/objects/cm.py:112]: IMO accepting license should be "safe", so method should be called only when `unaccepted` is the license status

**Review by Sealwing — APPROVED** [review]

**Review by a-alferov — APPROVED** [review]

**Pull request merged**